<a href="https://colab.research.google.com/github/nikhilnigam11/Netflix_movies_TV_Shows_cleaned/blob/main/Netflix_Movies_and_TV_show.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Nikhil Kumar Nigam**

Day 1 Task / Date 21 April 2025

Data Analyst Intern
# **Elevate Labs**

In [115]:
#importing python libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [116]:
df=pd.read_csv('/content/netflix.csv')
print('\033[1m-------------------Netflix Titles Data------------------\033[1m')
df.head()

-------------------Netflix Titles Data------------------


show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

In [117]:
#length of data
df.shape

(8807, 12)

In [118]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


# Total 8807 Rows and 12 columns in the Data Set

In [119]:
print('\033[1m<-----------Missing Values Percentage----------->\033[1m')

pd.DataFrame({'Total missing values':df.isna().sum(),
              'Percentage':round((df.isna().sum()/len(df))*100,2)})

<-----------Missing Values Percentage----------->


Total missing values  Percentage
show_id                          0        0.00
type                             0        0.00
title                            0        0.00
director                      2634       29.91
cast                           825        9.37
country                        831        9.44
date_added                      10        0.11
release_year                     0        0.00
rating                           4        0.05
duration                         3        0.03
listed_in                        0        0.00
description                      0        0.00

# **Missing Values**
1.   director column has 30 % missing values
2.   cast column has 9.3 % missing values
3.   country column has 9.4 % missing values

The missing values in the director column has 2634 NaN values which correspond with almost 30 percents of total data in that column. So, with just dropping the **NaN** values, we will lose lots of data, instead filling the NaN values with empty string is a good choice.

In [120]:
df.dropna(subset=['date_added','rating','duration'], inplace = True)

In [121]:
print('\033[1m<-----------Missing Values Percentage----------->\033[1m')
pd.DataFrame({'Total missing values':df.isna().sum(),
              'Percentage':round((df.isna().sum()/len(df))*100,2)})

<-----------Missing Values Percentage----------->


Total missing values  Percentage
show_id                          0        0.00
type                             0        0.00
title                            0        0.00
director                      2621       29.82
cast                           825        9.39
country                        829        9.43
date_added                       0        0.00
release_year                     0        0.00
rating                           0        0.00
duration                         0        0.00
listed_in                        0        0.00
description                      0        0.00

In [122]:
#replacing the NaN in director with the 'name not provided'

df.fillna({'director':'Director Name Not Provided'}, inplace=True)
df.head()

show_id     type                  title                    director  \
0      s1    Movie   Dick Johnson Is Dead             Kirsten Johnson   
1      s2  TV Show          Blood & Water  Director Name Not Provided   
2      s3  TV Show              Ganglands             Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans  Director Name Not Provided   
4      s5  TV Show           Kota Factory  Director Name Not Provided   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

In [123]:
#replacing the NaN in cast with the 'Name not provided'

df.fillna({'cast':'Name Not Provided'}, inplace=True)
df.head()

show_id     type                  title                    director  \
0      s1    Movie   Dick Johnson Is Dead             Kirsten Johnson   
1      s2  TV Show          Blood & Water  Director Name Not Provided   
2      s3  TV Show              Ganglands             Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans  Director Name Not Provided   
4      s5  TV Show           Kota Factory  Director Name Not Provided   

                                                cast        country  \
0                                  Name Not Provided  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                  Name Not Provided            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

In [124]:
print('\033[1m<-----------Missing Values Percentage----------->\033[1m')
pd.DataFrame({'Total missing values':df.isna().sum(),
              'Percentage':round((df.isna().sum()/len(df))*100,2)})

<-----------Missing Values Percentage----------->


Total missing values  Percentage
show_id                          0        0.00
type                             0        0.00
title                            0        0.00
director                         0        0.00
cast                             0        0.00
country                        829        9.43
date_added                       0        0.00
release_year                     0        0.00
rating                           0        0.00
duration                         0        0.00
listed_in                        0        0.00
description                      0        0.00

In [125]:
#number of unique values in our data
for i in df.columns:
  print(i,':',df[i].nunique())

show_id : 8790
type : 2
title : 8790
director : 4527
cast : 7679
country : 748
date_added : 1765
release_year : 74
rating : 14
duration : 220
listed_in : 513
description : 8758


# **Cleaning the Comma Separated Values in the director and cast column**

In [126]:
# Saperating the comma saperated values in the director column
df['director'] = df['director'].str.split(',')
df = df.explode('director')
df['director'] = df['director'].str.strip()
df.head()

show_id     type                  title                    director  \
0      s1    Movie   Dick Johnson Is Dead             Kirsten Johnson   
1      s2  TV Show          Blood & Water  Director Name Not Provided   
2      s3  TV Show              Ganglands             Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans  Director Name Not Provided   
4      s5  TV Show           Kota Factory  Director Name Not Provided   

                                                cast        country  \
0                                  Name Not Provided  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                  Name Not Provided            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

In [127]:
# Saperating the comma saperated values in the cast column
df['cast'] = df['cast'].str.split(',')
df = df.explode('cast')
df['cast'] = df['cast'].str.strip()
df.head()

show_id     type                 title                    director  \
0      s1    Movie  Dick Johnson Is Dead             Kirsten Johnson   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   

                cast        country          date_added  release_year rating  \
0  Name Not Provided  United States  September 25, 2021          2020  PG-13   
1         Ama Qamata   South Africa  September 24, 2021          2021  TV-MA   
1        Khosi Ngema   South Africa  September 24, 2021          2021  TV-MA   
1      Gail Mabalane   South Africa  September 24, 2021          2021  TV-MA   
1     Thabang Molaba   South Africa  September 24, 2021          2021  TV-MA   

    duration                                        listed_in  \
0     90 min                                    Documentaries   
1  2 Seasons  International TV Shows, TV Dramas, TV Mysteries   
1  2 Seasons  International TV Shows, TV Dramas, TV Mysteries   
1  2 Seasons  International TV Shows, TV Dramas, TV Mysteries   
1  2 Seasons  International TV Shows, TV Dramas, TV Mysteries   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...

In [128]:
dfn=df[df.duplicated()]
dfn.shape

(10, 12)

10 duplicate rows are found. Now Dropping them to get rid of duplicate values

In [129]:
df.drop_duplicates(inplace=True)

In [130]:
dfn=df[df.duplicated()]
dfn.shape

(0, 12)

Duplicate Rows dropped

In [131]:
# Saperating the comma saperated values in the 'listed_in' column
df['listed_in'] = df['listed_in'].str.split(',')
df = df.explode('listed_in')
df['listed_in'] = df['listed_in'].str.strip()
df.head()

show_id     type                 title                    director  \
0      s1    Movie  Dick Johnson Is Dead             Kirsten Johnson   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   

                cast        country          date_added  release_year rating  \
0  Name Not Provided  United States  September 25, 2021          2020  PG-13   
1         Ama Qamata   South Africa  September 24, 2021          2021  TV-MA   
1         Ama Qamata   South Africa  September 24, 2021          2021  TV-MA   
1         Ama Qamata   South Africa  September 24, 2021          2021  TV-MA   
1        Khosi Ngema   South Africa  September 24, 2021          2021  TV-MA   

    duration               listed_in  \
0     90 min           Documentaries   
1  2 Seasons  International TV Shows   
1  2 Seasons               TV Dramas   
1  2 Seasons            TV Mysteries   
1  2 Seasons  International TV Shows   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...

In [132]:
# Saperating the comma saperated values in the 'listed_in' column
df['country'] = df['country'].str.split(',')
df = df.explode('country')
df['country'] = df['country'].str.strip()
df.head()

show_id     type                 title                    director  \
0      s1    Movie  Dick Johnson Is Dead             Kirsten Johnson   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   

                cast        country          date_added  release_year rating  \
0  Name Not Provided  United States  September 25, 2021          2020  PG-13   
1         Ama Qamata   South Africa  September 24, 2021          2021  TV-MA   
1         Ama Qamata   South Africa  September 24, 2021          2021  TV-MA   
1         Ama Qamata   South Africa  September 24, 2021          2021  TV-MA   
1        Khosi Ngema   South Africa  September 24, 2021          2021  TV-MA   

    duration               listed_in  \
0     90 min           Documentaries   
1  2 Seasons  International TV Shows   
1  2 Seasons               TV Dramas   
1  2 Seasons            TV Mysteries   
1  2 Seasons  International TV Shows   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...

In [133]:
df['release_year'] = pd.to_datetime(df['release_year'], format='%Y')
df['release_year'] = df['release_year'].dt.year
df.head()

show_id     type                 title                    director  \
0      s1    Movie  Dick Johnson Is Dead             Kirsten Johnson   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   

                cast        country          date_added  release_year rating  \
0  Name Not Provided  United States  September 25, 2021          2020  PG-13   
1         Ama Qamata   South Africa  September 24, 2021          2021  TV-MA   
1         Ama Qamata   South Africa  September 24, 2021          2021  TV-MA   
1         Ama Qamata   South Africa  September 24, 2021          2021  TV-MA   
1        Khosi Ngema   South Africa  September 24, 2021          2021  TV-MA   

    duration               listed_in  \
0     90 min           Documentaries   
1  2 Seasons  International TV Shows   
1  2 Seasons               TV Dramas   
1  2 Seasons            TV Mysteries   
1  2 Seasons  International TV Shows   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...

# **leading spaces are observed in the date_added column data so to_datetime is giving error.**

In [134]:
df['date_added'] = df['date_added'].str.strip()
df.head()

show_id     type                 title                    director  \
0      s1    Movie  Dick Johnson Is Dead             Kirsten Johnson   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   

                cast        country          date_added  release_year rating  \
0  Name Not Provided  United States  September 25, 2021          2020  PG-13   
1         Ama Qamata   South Africa  September 24, 2021          2021  TV-MA   
1         Ama Qamata   South Africa  September 24, 2021          2021  TV-MA   
1         Ama Qamata   South Africa  September 24, 2021          2021  TV-MA   
1        Khosi Ngema   South Africa  September 24, 2021          2021  TV-MA   

    duration               listed_in  \
0     90 min           Documentaries   
1  2 Seasons  International TV Shows   
1  2 Seasons               TV Dramas   
1  2 Seasons            TV Mysteries   
1  2 Seasons  International TV Shows   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...

In [135]:
df['date_added']=pd.to_datetime(df['date_added'])
df.head()

show_id     type                 title                    director  \
0      s1    Movie  Dick Johnson Is Dead             Kirsten Johnson   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   

                cast        country date_added  release_year rating  \
0  Name Not Provided  United States 2021-09-25          2020  PG-13   
1         Ama Qamata   South Africa 2021-09-24          2021  TV-MA   
1         Ama Qamata   South Africa 2021-09-24          2021  TV-MA   
1         Ama Qamata   South Africa 2021-09-24          2021  TV-MA   
1        Khosi Ngema   South Africa 2021-09-24          2021  TV-MA   

    duration               listed_in  \
0     90 min           Documentaries   
1  2 Seasons  International TV Shows   
1  2 Seasons               TV Dramas   
1  2 Seasons            TV Mysteries   
1  2 Seasons  International TV Shows   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...

In [143]:
df.fillna({'country':'country Name Not Provided'}, inplace=True)
df.head()

show_id     type                 title                    director  \
0      s1    Movie  Dick Johnson Is Dead             Kirsten Johnson   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   
1      s2  TV Show         Blood & Water  Director Name Not Provided   

                cast        country date_added  release_year rating  \
0  Name Not Provided  United States 2021-09-25          2020  PG-13   
1         Ama Qamata   South Africa 2021-09-24          2021  TV-MA   
1         Ama Qamata   South Africa 2021-09-24          2021  TV-MA   
1         Ama Qamata   South Africa 2021-09-24          2021  TV-MA   
1        Khosi Ngema   South Africa 2021-09-24          2021  TV-MA   

    duration               listed_in  \
0     90 min           Documentaries   
1  2 Seasons  International TV Shows   
1  2 Seasons               TV Dramas   
1  2 Seasons            TV Mysteries   
1  2 Seasons  International TV Shows   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...  
1  After crossing paths at a party, a Cape Town t...

# **Cleaned Data ⬇**

In [145]:
df.sample(10)

show_id     type                                     title  \
2545   s2546    Movie                           Andaz Apna Apna   
3660   s3661  TV Show                           Unrequited Love   
2519   s2520  TV Show                Avatar: The Last Airbender   
6770   s6771  TV Show                           Flowering Heart   
4969   s4970    Movie                                  Birdshot   
5879   s5880    Movie                          The Ridiculous 6   
6433   s6434    Movie  Cats & Dogs: The Revenge of Kitty Galore   
7225   s7226    Movie                              Kraftidioten   
6982   s6983    Movie                                  Hokkabaz   
6473   s6474    Movie                   Chitty Chitty Bang Bang   

                        director                     cast         country  \
2545           Rajkumar Santoshi               Aamir Khan           India   
3660  Director Name Not Provided                  Shen Yu           China   
2519  Director Name Not Provided             Jack De Sena   United States   
6770  Director Name Not Provided               Yang Se-na     South Korea   
4969                 Mikhail Red            Dido Dela Paz     Philippines   
5879                Frank Coraci              Vanilla Ice   United States   
6433                 Brad Peyton           Kiernan Shipka       Australia   
7225          Hans Petter Moland  Birgitte Hjort Sørensen          Norway   
6982                  Cem Yılmaz               Ayçe Abana          Turkey   
6473                  Ken Hughes            Stanley Unwin  United Kingdom   

     date_added  release_year rating   duration                 listed_in  \
2545 2020-05-11          1994  TV-14    160 min               Cult Movies   
3660 2019-07-17          2019  TV-14   1 Season                 TV Dramas   
2519 2020-05-15          2007  TV-Y7  3 Seasons         Classic & Cult TV   
6770 2019-07-01          2016  TV-Y7   1 Season           Korean TV Shows   
4969 2018-03-26          2016  TV-MA    115 min      International Movies   
5879 2015-12-11          2015  TV-14    120 min                  Comedies   
6433 2019-02-01          2010     PG     82 min  Children & Family Movies   
7225 2017-01-05          2014  TV-MA    117 min                 Thrillers   
6982 2017-03-10          2006  TV-MA    118 min                    Dramas   
6473 2020-01-01          1968      G    146 min  Children & Family Movies   

                                            description  
2545  Broke and with no concrete plans for the futur...  
3660  A complicated, one-sided secret attraction sen...  
2519  Siblings Katara and Sokka wake young Aang from...  
6770  An ordinary student, who forms an advice club ...  
4969  A sheltered farm girl inadvertently shoots an ...  
5879  When his outlaw dad is kidnapped, Tommy “White...  
6433  Cats and dogs must set aside their differences...  
7225  When he sets out to avenge his son's murder, m...  
6982  Trying to resurrect his career, not-so-talente...  
6473  Quirky inventor Caractacus Potts and his famil...

In [146]:
print('\033[1m<-----------Missing Values Percentage----------->\033[1m')
pd.DataFrame({'Total missing values':df.isna().sum(),
              'Percentage':round((df.isna().sum()/len(df))*100,2)})

<-----------Missing Values Percentage----------->


Total missing values  Percentage
show_id                          0         0.0
type                             0         0.0
title                            0         0.0
director                         0         0.0
cast                             0         0.0
country                          0         0.0
date_added                       0         0.0
release_year                     0         0.0
rating                           0         0.0
duration                         0         0.0
listed_in                        0         0.0
description                      0         0.0